tests/subsys/fw_info/src/main.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <ztest.h>
+
+#include "fw_info.h"
+#include "pm_config.h"
+#include "flash.h"
+#include "misc/reboot.h"
+
+#define S1_ADDRESS 0x20000
+#define S1_SIZE 0x8000
+
+
+/* signature check of this will fail, causing .valid to be overwritten. */
+const struct fw_info dummy_s1 = {
+	.magic = {FIRMWARE_INFO_MAGIC},
+	.image_address = S1_ADDRESS,
+	.image_size = S1_SIZE,
+	.image_version = 0xFF,
+	.valid = CONFIG_FW_INFO_VALID_VAL,
+};
+
+u8_t fw_info_find_buf[0x1000 + sizeof(dummy_s1)];
+
+u32_t dummy_align;
+u8_t memeq_buf1[] = {1, 2, 3, 4, 5, 6, 7, 8}; /* Aligned */
+u8_t memeq_buf2[] = {1, 2, 3, 4, 5, 6, 7, 8}; /* Aligned */
+u8_t dummy_misalign;
```

Comment:
I don't know if this leads to an unaligned type behind it. Maybe try with byte-string buffers that are unaligned

Reply:
You were right, thanks! Fixed this now and added asserts for the expected alignment.